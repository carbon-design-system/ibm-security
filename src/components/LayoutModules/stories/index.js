/**
 * @file Story information.
 * @copyright IBM Security 2020 - 2021
 */

import { parameters, layoutModules } from '../../../../.storybook';

const LIBRARY = 'Library';

export default (title) => layoutModules(title);

export { LIBRARY, parameters };
