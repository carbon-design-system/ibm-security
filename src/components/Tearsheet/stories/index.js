/**
 * @file Tearsheet story guidance.
 * @copyright IBM Security 2020
 */

export default ({ displayName }) =>
  `To preserve animations when the '${displayName}' enters or leaves the DOM, use the 'isOpen' prop to control the component.`;
