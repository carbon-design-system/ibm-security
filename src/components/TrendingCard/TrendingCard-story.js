/**
 * @file Trending card stories.
 * @copyright IBM Security 2019 - 2021
 */

import React from 'react';

import { getDocsParameters, security } from '../../../.storybook';
import { TrendingCard } from '../..';

export default {
  title: security(TrendingCard.name),
  component: TrendingCard,
  args: {
    title: 'Title',
    href: '#',
    subtitle: 'Subtitle',
  },
  parameters: {
    ...getDocsParameters(),
  },
};

export const Default = args => <TrendingCard {...args} />;
