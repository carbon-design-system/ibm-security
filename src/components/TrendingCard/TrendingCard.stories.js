/**
 * @file Trending card stories.
 * @copyright IBM Security 2019
 */

import { breakpoints } from '@carbon/layout';

import { text } from '@storybook/addon-knobs';
import React from 'react';

import { components, meta } from '../../../.storybook';

import { TrendingCard } from '../..';

const props = () => ({
  title: text('Title (title)', 'Title'),
  href: text('Link (href)', '#'),
  subtitle: text('Subtitle (subtitle)', 'Subtitle'),
  style: { width: breakpoints.sm.width },
});

export const Default = ({ element }) => (
  <TrendingCard {...props()} element={element} />
);

export default meta(
  components('TrendingCard'),
  'Trending cards provide summary information of trending items with the ability to navigate to the details.'
);
