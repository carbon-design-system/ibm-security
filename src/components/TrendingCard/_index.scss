////
/// Trending card.
/// @group trending-card
/// @copyright IBM Security 2019
////

@import '@carbon/layout/scss/mini-unit';
@import '@carbon/layout/scss/spacing';
@import '@carbon/motion/scss/motion';
@import '@carbon/themes/scss/tokens';
@import '@carbon/type/scss/styles';

@import 'carbon-components/scss/globals/scss/helper-mixins';

@import '../../globals/motion/index';
@import '../../platform/body/mixins';
@import '../Component/mixins';

@include security--component($name: trending-card) {
  @include carbon--motion($name: standard, $mode: productive);

  display: block;
  min-width: carbon--mini-units($count: 24);
  padding: $carbon--spacing-05;
  background-color: $ui-01;
  transition-duration: $transition-duration--fast;
  text-decoration: unset;

  &:focus,
  &:hover {
    background-color: $hover-ui;
  }

  &:focus {
    @include focus-outline($type: outline);
  }

  &__title,
  &__subtitle {
    @include text-overflow;
  }

  &__title {
    @include carbon--type-style($name: body-long-01);

    margin-top: 0;
    margin-bottom: $carbon--spacing-02;
    color: $body__color__text;
  }

  &__subtitle {
    @include carbon--type-style($name: caption-01);
    color: $text-02;
  }
}
