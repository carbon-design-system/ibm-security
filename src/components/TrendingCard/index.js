/**
 * @file Trending card.
 * @copyright IBM Security 2019
 */

import classnames from 'classnames';
import { elementType, node, string } from 'prop-types';
import React, { Fragment } from 'react';

import { getComponentNamespace } from '../../globals/namespace';

const namespace = getComponentNamespace('trending-card');

/**
 * Trending cards provide summary information of trending items with the ability to navigate to the details.
 */
const TrendingCard = ({ className, element, title, subtitle, ...other }) =>
  React.createElement(
    element,
    { className: classnames(namespace, className), ...other },
    <Fragment>
      <h3 className={`${namespace}__title`}>{title}</h3>
      {subtitle && <span className={`${namespace}__subtitle`}>{subtitle}</span>}
    </Fragment>
  );

TrendingCard.defaultProps = {
  subtitle: null,
  element: 'a',
  className: null,
};

TrendingCard.propTypes = {
  /** Specify the text of the title */
  title: string.isRequired,

  /** Specify the content of the subtitle */
  subtitle: node,

  /** Specify the base element to use to build the link. This can also accept custom components like `Link` from `react-router` */
  element: elementType,

  /** Provide an optional class to be applied to the containing node */
  className: string,
};

export default TrendingCard;
