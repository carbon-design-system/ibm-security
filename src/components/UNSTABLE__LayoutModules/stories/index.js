/**
 * @file Story information.
 * @copyright IBM Security 2020
 */

import { patterns } from '../../../../.storybook';

export default ({ displayName, name }) =>
  patterns(`UNSTABLE Layout Modules/${displayName || name}`);
