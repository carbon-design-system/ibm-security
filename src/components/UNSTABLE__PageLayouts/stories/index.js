/**
 * @file Story information.
 * @copyright IBM Security 2020 - 2021
 */

import { Grid } from 'carbon-components-react';
import { layout04 } from '@carbon/layout';
import classnames from 'classnames';
import React from 'react';

import { disableCentered } from '../../../../.storybook';

import {
  CarbonHeader,
  HeaderName,
  InlineNotification,
  NotificationActionButton,
} from '../../..';

export default {
  parameters: {
    ...disableCentered(),
  },
  decorators: [
    story => (
      <>
        <div style={{ height: layout04 }}>
          <CarbonHeader aria-label="IBM Security">
            <HeaderName prefix="IBM">Security</HeaderName>
          </CarbonHeader>
        </div>

        <InlineNotification
          className="page-layouts__banner"
          actions={
            <NotificationActionButton
              href="https://github.com/carbon-design-system/carbon/issues/7717"
              rel="noopener noreferrer"
              target="_blank"
            >
              More info
            </NotificationActionButton>
          }
          kind="info"
          subtitle="Page layouts will remain Canary until the 2021 Carbon release."
          title=""
          hideCloseButton
        />

        <Grid>{story()}</Grid>
      </>
    ),
  ],
};

// TODO: Remove workaround for https://github.ibm.com/security/design-core-experience/issues/241
export const withContainer = WrappedComponent => ({ className, ...other }) => (
  <WrappedComponent
    className={classnames('container--narrow', className)}
    {...other}
  />
);
