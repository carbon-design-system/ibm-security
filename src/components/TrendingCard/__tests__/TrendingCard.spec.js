/**
 * @file Trending card tests.
 * @copyright IBM Security 2019
 */

import { render } from '@testing-library/react';
import React from 'react';

import { Default as TrendingCard } from '../TrendingCard.stories';

describe('TrendingCard', () => {
  it('renders', () => {
    expect(render(<TrendingCard />).container.firstChild).toMatchSnapshot();
  });

  it('renders a custom link', () => {
    expect(
      render(<TrendingCard element="Link" />).container.firstChild
    ).toMatchSnapshot();
  });
});
