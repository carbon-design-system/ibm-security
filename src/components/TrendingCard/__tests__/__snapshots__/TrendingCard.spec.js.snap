// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendingCard renders 1`] = `
<a
  class="security--trending-card"
  href="#"
  style="width: 20rem;"
>
  <h3
    class="security--trending-card__title"
  >
    Title
  </h3>
  <span
    class="security--trending-card__subtitle"
  >
    Subtitle
  </span>
</a>
`;

exports[`TrendingCard renders a custom link 1`] = `
<link
  class="security--trending-card"
  href="#"
  style="width: 20rem;"
>
  <h3
    class="security--trending-card__title"
  >
    Title
  </h3>
  <span
    class="security--trending-card__subtitle"
  >
    Subtitle
  </span>
</link>
`;
